mesonbuild/backend/vs2010backend.py

```diff
@@ -129,6 +129,14 @@ def get_non_primary_lang_intellisense_fields(vslite_ctx: dict,
             defs_paths_opts_per_lang_and_buildtype[src_lang][buildtype] = Vs2010Backend._extract_nmake_fields(args_list)
     return defs_paths_opts_per_lang_and_buildtype
 
+# Returns if a target generates a manifest or not.
+def get_gen_manifest(target):
+    if isinstance(target, build.BuildTarget):
+        upper_args = [arg.upper() for arg in target.link_args]
+        manifest_args = [arg for arg in upper_args if arg == '/MANIFEST' or arg.startswith('/MANIFEST:')]
+        return len(manifest_args) == 0 or manifest_args[-1] != '/MANIFEST:NO'
```

Comment:
does `/MANIFEST` have to be the last argument, or could there be another argument after that?

Reply:
`/MANIFEST`, `/MANIFEST:EMBED`, and `/MANIFEST:EMBED,ID=xxx` can enable the manifest creation.
I got the same error as #12472 when ignoring `/MANIFEST:EMBED` in the `get_gen_manifest` function.

Reply:
My concern is that if your link args are:
```python
['/somearg', '/MANIFEST:NO', '/someother-arg']
```
that this will fail. And (assumine that link.exe is sane):
```python
['/somearg', '/MANIFEST:NO', '/MANIFEST', '/someother-arg']
```
makes this more complicated. I think we really need to do something like:
```python
for x in reversed(manifest_args):
    if x == '/MANIFEST:NO':
        break
    if x.startswith('/MANIFEST'):
        return False
return True
```

So that we look at the last occurance of the `/MANIFEST` argument, and properly handle it being not the last argument in the link_args

Reply:
He already filtered non-MANIFEST args in the list comprehension, so I think the proposed code is ok. But I thought about proposing a similar explicit for loop instead of list comprehensions. I think it would be more efficient, because it can stop at the first occurence found.

Reply:
You're right, I missed that somehow. derp.

Yeah, I think I'd prefer a single loop instead of the two comprehensions as well.

Reply:
So, should it be like this?

```python
# Returns if a target generates a manifest or not.
def get_gen_manifest(target):
    if isinstance(target, build.BuildTarget):
        for arg in reversed(target.link_args):
            arg = arg.upper()
            if arg == '/MANIFEST:NO':
                return False
            if arg == '/MANIFEST' or arg.startswith('/MANIFEST:'):
                break
    return True
```

`if arg.startswith('/MANIFEST'):` doesn't work because there are other options that start with `/MANIFEST`.